tests/BaseApiControllerTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Saritasa\LaravelControllers\Tests;
+
+use Dingo\Api\Http\Request;
+use Illuminate\Contracts\Validation\Factory;
+use Illuminate\Contracts\Validation\Validator;
+use Illuminate\Support\MessageBag;
+use Mockery;
+use Mockery\MockInterface;
+use Saritasa\DingoApi\Exceptions\ValidationException;
+use Saritasa\LaravelControllers\Api\BaseApiController;
+use Saritasa\Transformers\IDataTransformer;
+
+/**
+ * Tests for base api controller.
+ */
+class BaseApiControllerTest extends TestCase
+{
+    /**
+     * Data transformer mock.
+     *
+     * @var MockInterface|IDataTransformer
+     */
+    protected $transformer;
+
+    /**
+     * Base api controller.
+     *
+     * @var MockInterface|BaseApiController
+     */
+    protected $baseApiController;
+
+    /**
+     * Default request instance.
+     *
+     * @var MockInterface|Request
+     */
+    protected $request;
+
+    /**
+     * Prepare tests for run.
+     *
+     * @return void
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->transformer = Mockery::mock(IDataTransformer::class);
+        $this->baseApiController = Mockery::mock(BaseApiController::class)
+            ->makePartial()
+            ->shouldAllowMockingProtectedMethods();
+
+        $this->request = Mockery::mock(Request::class);
+    }
+
+    /**
+     * Test validation method when success.
```

Comment:
*when data is valid*. Validation can be success even when data is invalid (no error occurred during validate attempt). Just like in js: 
```js
this.$validateAll()
    .then(validationResult => {
        // validation succes but result can be false
    })
   .catch(validationFailureError => {
        / /validation failure (maybe some invalid rules was used)
    });
```